src/products/components/ProductStocks/ProductStocks.tsx

```diff
@@ -131,13 +193,23 @@ const ProductStocks: React.FC<ProductStocksProps> = ({
   const intl = useIntl();
   const anchor = React.useRef<HTMLDivElement>();
   const [isExpanded, setExpansionState] = React.useState(false);
+  const unitsLeft = data.globalThreshold - data.globalSoldUnits;
 
   const warehousesToAssign =
     warehouses?.filter(
       warehouse => !stocks.some(stock => stock.id === warehouse.id)
     ) || [];
   const formErrors = getFormErrors(["sku"], errors);
 
+  const onThresholdChange = createNonNegativeValueChangeHandler(
+    onFormDataChange
+  );
+
+  const onDateChange = createFutureDateChangeHandler(
+    "preorderEndDateTime",
+    onFormDataChange
+  );
+
   return (
     <Card>
```

Comment:
I think since almost all the content in this component is now separated by preorder logic, there should be two separate components for this.
- Make ProductStocksCard and put the Card, CardTitle, TextField And Checkbox in there
- Make PreorderProductStocks and ProductStocks components, that'll do
```
<ProductStocksCard onChange={change} /* otherProps etc. */ >
// All the preorder content here
</ProductStocksCard>